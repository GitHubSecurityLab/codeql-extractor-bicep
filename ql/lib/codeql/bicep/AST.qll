import ast.AstNodes
import ast.Literals
import ast.Conditionals
import ast.Calls
import ast.Expr
import ast.Stmts
import ast.Loops
import ast.Misc
import ast.Idents
import ast.Variables
import ast.Resources
import ast.Types
