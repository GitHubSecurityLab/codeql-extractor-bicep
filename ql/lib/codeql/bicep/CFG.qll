/** Provides classes representing the control flow graph. */

import codeql.Locations
import controlflow.BasicBlocks
import controlflow.ControlFlowGraph
import controlflow.CfgNodes as CfgNodes