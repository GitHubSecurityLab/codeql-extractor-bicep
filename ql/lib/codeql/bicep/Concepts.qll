private import codeql.bicep.AST
private import codeql.bicep.CFG

/**
 *  A Public Resource is a resource that is publicly accessible to the Internet.
 */
abstract class PublicResource extends Resource {
  /**
   *  Returns the property that indicates public access.
   */
  abstract Expr getPublicAccessProperty();
}

module Cryptography {
  abstract class WeakTlsVersion extends Resource {
    abstract Expr getWeakTlsVersionProperty();

    /**
     *  Returns true if the resource has a weak TLS version.
     * 
     *  1.0 and 1.1 are considered weak TLS versions.
     */
    predicate hasWeakTlsVersion() {
      exists(StringLiteral literal |
        literal = this.getWeakTlsVersionProperty() and
        literal.getValue().regexpMatch("^(1\\.0|1\\.1)$")
      )
    }
  }
}
