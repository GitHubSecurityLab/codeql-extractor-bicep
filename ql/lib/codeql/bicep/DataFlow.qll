import dataflow.Ssa
import dataflow.DataFlow
