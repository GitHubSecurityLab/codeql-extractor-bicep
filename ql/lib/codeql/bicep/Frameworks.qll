import frameworks.Microsoft.Compute
import frameworks.Microsoft.Network
import frameworks.Microsoft.Storage
import frameworks.Microsoft.Databases
import frameworks.Microsoft.KeyVault