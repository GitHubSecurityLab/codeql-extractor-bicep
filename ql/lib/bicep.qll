import codeql.Locations
import codeql.files.FileSystem
// AST
import codeql.bicep.AST
// CFG
import codeql.bicep.CFG
// Frameworks
import codeql.bicep.Frameworks
import codeql.bicep.Concepts
