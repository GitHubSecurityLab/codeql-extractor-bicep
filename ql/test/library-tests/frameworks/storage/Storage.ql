import bicep

query predicate storage(Storage::StorageAccounts storageAccount) {
  any() 
}
