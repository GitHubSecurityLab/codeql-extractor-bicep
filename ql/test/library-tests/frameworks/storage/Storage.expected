| app.bicep:2:1:10:1 | ResourceDeclaration |
| app.bicep:13:1:34:1 | ResourceDeclaration |
| app.bicep:37:1:48:1 | ResourceDeclaration |
